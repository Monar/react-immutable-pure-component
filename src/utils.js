import { is, getIn } from 'immutable';

export function check(config, prev = {}, next = {}, label) {
  let checkItem = createChecker(prev, next, label);
  let checklist = config || Object.keys({ ...next, ...prev });

  return !checklist.every(checkItem);
}

function createChecker(prev, next, checkName) {
  return function(name) {
    if (typeof name === 'string') {
      return is(next[name], prev[name]);
    }

    if (!getIn) {
      let value = JSON.stringify(name);
      throw new TypeError(
        `Not supported value "${value}" provided to ${checkName}, try updating immutable to v4`,
      );
    }

    return is(getIn(next, name), getIn(prev, name));
  };
}
