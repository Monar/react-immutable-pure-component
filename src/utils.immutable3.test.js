/* eslint-env node, jest */
import { check } from './utils';

jest.mock('immutable', () => ({ is: () => true }));

describe('Test check when immutable does not have getIn', () => {
  it('should throw when item to check is not a string', () => {
    let config = [['should', 'not', 'work'], 'this', "doesn't", 'matter'];
    expect(() => check(config, {}, {}, 'test')).toThrowErrorMatchingSnapshot();
  });
});
