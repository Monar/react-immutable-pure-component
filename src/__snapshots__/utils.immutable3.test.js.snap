// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test check when immutable does not have getIn should throw when item to check is not a string 1`] = `"Not supported value \\"[\\"should\\",\\"not\\",\\"work\\"]\\" provided to test, try updating immutable to v4"`;
